src/video/vga_palette.c

```diff
@@ -8,24 +8,26 @@ void vga_palette_init(vga_palette *palette) {
     memset(palette, 0, sizeof(vga_palette));
 }
 
-void vga_palette_tint_range(vga_palette *pal, vga_index ref_index, vga_index start, vga_index end, uint8_t step) {
+void vga_palette_tint_range(vga_palette *pal, vga_index ref_index, vga_index first, vga_index last, uint8_t step) {
     vga_color ref = pal->colors[ref_index];
     uint32_t m, u;
-    for(vga_index i = start; i < end; i++) {
+    vga_index i = first;
+    do {
         m = max3(pal->colors[i].r, pal->colors[i].g, pal->colors[i].b);
         u = (step * m) >> 8;
         pal->colors[i].r += u * (ref.r - pal->colors[i].r) >> 8;
         pal->colors[i].g += u * (ref.g - pal->colors[i].g) >> 8;
         pal->colors[i].b += u * (ref.b - pal->colors[i].b) >> 8;
-    }
+    } while(i++ != last);
```

Comment:
Why the do-while loop ? For-loop seems like a clearer option (?)

Reply:
switched back to exclusive range, with the for loop.